lib/nxt_support/preprocessors/downcase_preprocessor.rb

```diff
@@ -0,0 +1,18 @@
+module NxtSupport
+  module Preprocessors
+    class DowncasePreprocessor
+      attr_accessor :value
+
+      def initialize(value)
+        @value = value
+      end
+
+      def call
+        return value unless value.is_a?(String)
```

Comment:
I would not do that but rather raise an error if the value does not respond to :downcase!